sdk/core/azure-core/src/main/java/com/azure/core/util/paging/SyncPageRetriever.java

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.paging;
+
+import com.azure.core.util.IterableStream;
+
+/**
+ * This class handles retrieving pages.
+ *
+ * @param <C> Type of the continuation token.
+ * @param <P> the page elements type
+ */
+@FunctionalInterface
+public interface SyncPageRetriever<C, P> {
```

Comment:
Should this be named `PageRetrieverSync`? We don't have `Sync` as a prefix for sync `HttpPipelinePolicy`.